sdk/rust/oak/src/io/mod.rs

```diff
@@ -16,35 +16,22 @@
 
 //! Wrappers for Oak SDK types to allow their use with [`std::io`].
 
-use crate::{channel_close, channel_write, OakStatus};
+use crate::{channel_write, OakStatus};
 use std::io;
 
-/// Wrapper for a WriteHandle to implement the [`std::io::Write`] trait.
-///
-/// Methods taking `Write` trait objects require mutable references,
-/// so use a distinct type rather than implementing the trait on the
-/// base [`WriteHandle`] type.
-///
-/// [`WriteHandle`]: crate::WriteHandle
-#[derive(Debug, Copy, Clone, PartialEq)]
-pub struct Channel {
-    pub handle: crate::WriteHandle,
-}
+mod decodable;
+mod encodable;
+mod receiver;
+mod sender;
 
-impl Channel {
-    /// Create a new `io::Channel` that uses the given channel handle.
-    pub fn new(handle: crate::WriteHandle) -> Self {
-        Channel { handle }
-    }
-    /// Close the underlying channel handle.
-    pub fn close(self) -> std::io::Result<()> {
-        channel_close(self.handle.handle).map_err(error_from_nonok_status)
-    }
-}
+pub use decodable::Decodable;
+pub use encodable::Encodable;
+pub use receiver::Receiver;
+pub use sender::Sender;
 
-/// Implement the [`std::io::Write`] trait for `io::Channel`, to allow logging
+/// Implement the [`std::io::Write`] trait for `io::Sender`, to allow logging
 /// and use of protobuf serialization methods.
-impl std::io::Write for Channel {
+impl std::io::Write for Sender {
```

Comment:
nit: should this move into `receiver.rs`?

Reply:
Fixed in #499.